java-frontend/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java

```diff
@@ -233,7 +234,16 @@ public void accept(TreeVisitor visitor) {
 
   @Override
   public int getLine() {
-    return parameters.openParenToken().getLine();
+    InternalSyntaxToken token = parameters.openParenToken();
+    if(token != null){
+      return token.getLine();
+    }else{
+      //type cast may fail, it is fine. We will just add a new case if it happens.
+      //could first try with type cast and fallback parameters
+      //but cannot reach full coverage
+      InternalSyntaxToken name = (InternalSyntaxToken)simpleName().identifierToken();
+      return name.getLine();
+    }
```

Comment:
We should format the code to make it more readable

Reply:
:+1: